### PR TITLE
Add script tag support

### LINKED ISSUE #15: Proposal: Use default contao template fields
We should use replaceDynamicScriptTags Hook in combination with [[TL_BODY]], [[TL_CSS]], [[TL_HEAD]] to insert encore assets instead of using our own template variables. So this bundle would be easier to use and need no changes to the default fe_page template.

Possible Roadmap:
1. add ´use_contao_script_tags` config with default value false
2. if set to true, these will be used instead of our own template variables
3. in version 2.0 this option will set to true by default (or removed in combination with our own template tags)

Disadvantage:
- the hook just pass the buffer, no other context variables -> in contao 4.9 we have the page object in the request, so we could use this. For 4.4 we could drop the support in version 2.0 or add the page object before (if not already there, this must be checked)

Links:
https://docs.contao.org/dev/reference/hooks/replaceDynamicScriptTags/

### PR DESCRIPTION
**Pull request by koertho**
This PR add script tags like contao do. This fixes the problem, that some entries added after the generatePage hook.

- use script tags like contao do (fix issue with GeneratePage hook runs to early)
- add use_contao_template_variables option to use contao template variables instead of custom variables introduced with this bundle (fixes #15)
- updates the readme

### CODE REVIEW

**Review by Defcon0 — APPROVED** [review]
Looks good :-) Thanks for fixing! I'll test it right away.

**Comment by koertho** [comment]
@Defcon0 Thanks! Last point is updating the docs.

File: src/EventListener/GeneratePageListener.php [hunk 1]
```diff
@@ -60,7 +60,9 @@ public function __invoke(PageModel $pageModel, LayoutModel $layout, PageRegular
         if (!$layout->addEncore) {
             return;
         }
-        $this->addEncore($pageModel, $layout, $pageRegular);
+        if (!isset($this->bundleConfig['use_contao_template_variables']) || true !== $this->bundleConfig['use_contao_template_variables']) {
```
Comment (AlexejKossmann) [inline src/EventListener/GeneratePageListener.php:63]: why you ask 

true !== $this->bundleConfig['use_contao_template_variables']

and not just !$this->bundleConfig['use_contao_template_variables']  or false ===

Comment (koertho) [reply src/EventListener/GeneratePageListener.php:63]: Cause it should only be applied if it's an boolean true ;)

**Review by AlexejKossmann — APPROVED** [review]

**Pull request merged**